src/rpc/net.cpp

```diff
@@ -746,8 +746,121 @@ UniValue Aggregate(const std::vector<DimensionType>& dimensions, const NetStats:
+                 {RPCResult::Type::OBJ_DYN,
+                  "keys for the dimension type",
+                  true,
+                  "Network traffic statistics.\n",
+                  net_stats::NetMsgStat()}}},
+            // A return object with two levels of data: two dimensions
+            RPCResult{
+                RPCResult::Type::OBJ_DYN,
+                "",
+                "",
+                {{RPCResult::Type::OBJ_DYN,
+                  "keys for the first dimension type",
+                  "Network traffic statistics.\n",
+                  {{RPCResult::Type::OBJ_DYN,
+                    "keys for the second dimension type",
+                    "",
+                    net_stats::NetMsgStat()}}}}},
+            // A return object with three levels of data: three dimensions
+            RPCResult{
+                RPCResult::Type::OBJ_DYN,
+                "",
+                "",
+                {{RPCResult::Type::OBJ_DYN,
+                  "keys for the first dimension type",
+                  "Network traffic statistics.\n",
+                  {{RPCResult::Type::OBJ_DYN,
+                    "keys for the second dimension type",
+                    "",
+                    {{RPCResult::Type::OBJ_DYN,
+                      "keys for the third dimension type",
+                      "",
+                      net_stats::NetMsgStat()}}}}}}},
+            // A return object with four levels of data: four dimensions
+            RPCResult{
+                RPCResult::Type::OBJ_DYN,
+                "",
+                "",
+                {{RPCResult::Type::OBJ_DYN,
+                  "keys for the first dimension type",
+                  "Network traffic statistics.\n",
+                  {{RPCResult::Type::OBJ_DYN,
+                    "keys for the second dimension type",
+                    "",
+                    {{RPCResult::Type::OBJ_DYN,
+                      "keys for the third dimension type",
+                      "",
+                      {{RPCResult::Type::OBJ_DYN,
+                        "keys for the fourth dimension type",
+                        "",
+                        net_stats::NetMsgStat()}}}}}}}}}},
+
+        RPCExamples{
+            HelpExampleCli("getnetmsgstats", "") +
+            HelpExampleCli("getnetmsgstats", R"('["connection_type","message_type"]')") +
+            HelpExampleRpc("getnetmsgstats", "") +
+            HelpExampleRpc("getnetmsgstats", R"(["message_type","network"])")},
+        /*fun=*/[&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            NodeContext& node = EnsureAnyNodeContext(request.context);
+            const CConnman& connman = EnsureConnman(node);
+            std::vector<net_stats::DimensionType> dimensions;
+
+            // Aggregate the messages to show only requested dimensions
+            if (request.params[0].isArray()) {
+                const UniValue raw_dimensions = request.params[0].get_array();
```

Comment:
Would be nice to have some validation of user input here - `getnetmsgstats '["foo","bar"]'` could return an error message, listing the supported dimensions.